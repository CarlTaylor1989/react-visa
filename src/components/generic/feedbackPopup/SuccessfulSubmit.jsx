import React from 'react';
import { Translate } from 'react-redux-i18n';
import PropTypes from 'prop-types';

const SuccessfulSubmit = ({ closePopup }) => (
  <>
    <div className="popupText">
      <Translate value="generic.feedback.popupSuccesText" />
    </div>
    <div className="buttons">
      <button
        type="button"
        className="closeBtn"
        onClick={closePopup}
      >
        <Translate className="text" value="generic.feedback.closeBtn" />
      </button>
    </div>
  </>
);

SuccessfulSubmit.propTypes = {
  closePopup: PropTypes.func.isRequired,
};

export default SuccessfulSubmit;
