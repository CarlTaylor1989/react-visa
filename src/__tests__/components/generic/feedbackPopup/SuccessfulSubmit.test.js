/* global toJson */
import React from 'react';
import SuccessfulSubmit from '../../../../components/generic/feedbackPopup/SuccessfulSubmit';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    closePopup: jest.fn()
  };
});

test('renders the SuccessfulSubmit component', () => {
  const wrapper = shallow(<SuccessfulSubmit {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
