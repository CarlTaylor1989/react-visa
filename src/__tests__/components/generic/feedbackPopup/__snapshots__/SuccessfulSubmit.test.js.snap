// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SuccessfulSubmit component 1`] = `
<Fragment>
  <div
    className="popupText"
  >
    <Translate
      tag="span"
      value="generic.feedback.popupSuccesText"
    />
  </div>
  <div
    className="buttons"
  >
    <button
      className="closeBtn"
      onClick={[MockFunction]}
      type="button"
    >
      <Translate
        className="text"
        tag="span"
        value="generic.feedback.closeBtn"
      />
    </button>
  </div>
</Fragment>
`;
